client/src/lib/CKANApi/DataStore/Query.js

```diff
@@ -227,12 +229,18 @@ class Query {
       ? ` WHERE (${this.filterBundles.map(bundle =>
         // Map those bundles - set "column" ILIKE '%field%' (and escape single quotes)
         bundle.map(({ column, term, strict, match }) => {
-          const queryColumn = column.replace('"', '""');
+          const columnName = column.replace('"', '""');
+          // Check for defined field types, and perform casting if necessary
+          const columnFieldType = this.fieldTypes.filter(fieldType => fieldType.label === column);
+          const queryColumn = columnFieldType.length && columnFieldType[0].type === 'numeric'
```

Comment:
What do you think about making this `type !== 'text'` (or whatever that type is).

Reply:
Sure